tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -0,0 +1,260 @@
+                    print(
+                        f"Processing Model: {GMM.name} for {tect_type}, {GMM_idx + 1} / {len(GMMs)}"
+                    )
+                    with imdb as imdb:
+                        for site in sites:
+                            rupture_context_df = create_rupture_context_df(
+                                fault_df.merge(
+                                    distance_store.station_data(site),
+                                    left_on="fault_name",
+                                    right_index=True,
+                                ),
+                                site_df.loc[site],
+                                nhm_data,
+                                classdef.TectType[tect_type],
+                            )
+                            gmm_calculated_df = openquake_wrapper_vectorized.oq_run(
+                                GMM,
+                                classdef.TectType[tect_type],
+                                rupture_context_df,
+                                str(im),
+                                psa_periods if im is gc.im.IMType.pSA else None,
+                            )
+                            # Matching the index with rupture_df
+                            # to have a right rupture label
+                            gmm_calculated_df.set_index(
+                                rupture_df[
+                                    rupture_df["rupture_name"].isin(
+                                        rupture_context_df["rupture_name"]
+                                    )
+                                ].index,
+                                inplace=True,
+                            )
+
+                            # Relabel the columns
+                            # PGA_mean -> PGA
+                            gmm_calculated_df.columns = [
+                                f"{'_'.join(col.split('_')[:2]) if im is gc.im.IMType.pSA else im}"
+                                if col.endswith("mean")
+                                else col
+                                for col in gmm_calculated_df
+                            ]
+                            # PGA_std_Total -> PGA_sigma
+                            gmm_calculated_df.columns = [
+                                f"{'_'.join(col.split('_')[:2]) if im is gc.im.IMType.pSA else im}_sigma"
+                                if col.endswith("std_Total")
+                                else col
+                                for col in gmm_calculated_df
+                            ]
+
+                            # Write an im_df to the given station/site
+                            imdb.add_im_data(
+                                site,
+                                gmm_calculated_df.loc[
+                                    :,
+                                    # Only mean and std_Total are needed
+                                    ~gmm_calculated_df.columns.str.endswith(
```

Comment:
Instead of masking here, you could drop those two columns after you get gmm_calculated_df back from the wrapper. Just a thought.

Reply:
Tried to look up the ways of dropping columns with multiple conditions but it all ended up masking was the simpler/only option I could find.

But minor update, it turns out pandas' string methods `endswith` only works with one pattern, so replace it with `contains`